### PR TITLE
SignatureSha256 and PackageFamilyName

### PR DESCRIPTION
**Pull request by OfficialEsco**
Resolves https://github.com/microsoft/winget-pkgs/issues/24066

### CODE REVIEW

**Comment by OfficialEsco** [comment]
@vedantmgoyal2009

**Comment by vedantmgoyal2009** [comment]
Thanks, now I will add some more things. Actually, I don't know about YamlParsing which is why I could not proceed.

File: Tools/YamlCreate.ps1 [hunk 1]
```diff
@@ -197,6 +199,7 @@ Function Read-WinGet-InstallerValues {
             $InstallerSha256 = (Get-FileHash -Path $dest -Algorithm SHA256).Hash
             $FileInformation = Get-AppLockerFileInformation -Path $dest | Select-Object Publisher | Select-String -Pattern "{[A-Z0-9]{8}-([A-Z0-9]{4}-){3}[A-Z0-9]{12}}"
             $MSIProductCode = $FileInformation.Matches
+            if (Get-Command 'winget.exe' -ErrorAction SilentlyContinue) { $SignatureSha256 = winget hash -m $dest | Select-String -Pattern "SignatureSha256:" | ConvertFrom-String; if ($SignatureSha256.P2) { $SignatureSha256 = $SignatureSha256.P2.ToUpper() }}
```
Comment (vedantmgoyal2009) [inline Tools/YamlCreate.ps1:202]: Can't we add a condition like this?
```
if (($dest.EndsWith('appx') -or $dest.EndsWith('msix') -or $dest.EndsWith('appxbundle') -or $dest.EndsWith('msixbundle')) -and Get-Command 'winget.exe' -ErrorAction SilentlyContinue)  { 
winget hash -m $dest | Select-String -Pattern "SignatureSha256:" | ConvertFrom-String
if ($SignatureSha256.P2) { $SignatureSha256 = $SignatureSha256.P2.ToUpper() }
}
```

**Comment by vedantmgoyal2009** [comment]
Once this will get merged, I will do some more chnages. Thanks for this. I do not know about YamlParsing, so I couldn't proceed.

File: Tools/YamlCreate.ps1 [hunk 2]
```diff
@@ -256,6 +259,24 @@ Function Read-WinGet-InstallerValues {
...
+    }
+
+    if ($InstallerType -ieq 'msix' -or $InstallerType -ieq 'appx') {
+        do {
+            Write-Host
+            Write-Host -ForegroundColor 'Yellow' -Object '[Recommended] Enter the installer PackageFamilyName'
+            $PackageFamilyName = Read-Host -Prompt 'PackageFamilyName' | TrimString
+        } while (-not [string]::IsNullOrWhiteSpace($PackageFamilyName) -and ($PackageFamilyName.Length -gt $InstallerSchema.definitions.PackageFamilyName.maxLength))
+    }
+
```
Comment (vedantmgoyal2009) [inline Tools/YamlCreate.ps1:279]:
```suggestion
    if ($InstallerType -ieq 'msix' -or $InstallerType -ieq 'appx') {
        if ([string]::IsNullOrWhiteSpace($SignatureSha256)) {
            do {
                Write-Host
                Write-Host -ForegroundColor 'Yellow' -Object '[Recommended] Enter the installer SignatureSha256'
                $SignatureSha256 = Read-Host -Prompt 'SignatureSha256' | TrimString
            } while (-not [string]::IsNullOrWhiteSpace($SignatureSha256) -and ($SignatureSha256 -notmatch $InstallerSchema.definitions.Installer.properties.SignatureSha256.pattern))
        }
        do {
            Write-Host
            Write-Host -ForegroundColor 'Yellow' -Object '[Recommended] Enter the installer PackageFamilyName'
            $PackageFamilyName = Read-Host -Prompt 'PackageFamilyName' | TrimString
        } while (-not [string]::IsNullOrWhiteSpace($PackageFamilyName) -and ($PackageFamilyName.Length -gt $InstallerSchema.definitions.PackageFamilyName.maxLength))
}

```

**Review by vedantmgoyal2009 — CHANGES_REQUESTED** [review]
If we are checking the same condition, why two times...

File: Tools/YamlCreate.ps1 (hunk 1 above)
Comment (Trenly) [reply Tools/YamlCreate.ps1:202]: I’m not sure. If I’m not mistaken, the reason we want to check regardless of the file extension is that some msix installers can be packaged inside an exe wrapper

**Review by Trenly — APPROVED** [review]

**Pull request merged**